### ISSUE TITLE
Load configuration from a specific file - is it possible?

### ISSUE THREAD

**Issue by vnnv** [avaje/avaje-config#129]
I didn't find a way to crate a `Configuration` using a specific file via constructor or a specific method.
Is there a way to load an yaml file containing configuration that is stored somewhere on non default location? I saw that I can pass file name using `pros.file` system property or env variable, but I'd like something like `Config.from(configFileName)` ?

My use case is I have a single properties file that is application configuration but also ebean configuration. I'd like to use yaml and achieve the same.

**Comment by SentryMan** [comment]
> I'd like something like `Config.from(configFileName)`?

what benefit does this get you over the configuration-based approach?

**Comment by vnnv** [comment]
> what benefit does this get you over the configuration-based approach?

Full control what is loaded for example. Btw, I tried with `load.properties`. Not sure why it is not working. 
I use such code:
```
        System.setProperty("load.properties", fileName);
        final Configuration configuration = Config.asConfiguration();
        final Properties properties = Config.asProperties();
```
Properties is almost empty - 2 keys only. Also add it via jvm arguments - `-D` style also did not worked.

**Comment by SentryMan** [comment]
> Full control what is loaded for example.

you still have that control doing it via configuration.

> it is not working.

is the property file on the classpath? if it's somewhere else on your system you'll need to use the file path instead of the name.

**Comment by SentryMan** [comment]
Ah I see, the issue is a bug on our side

**Comment by vnnv** [comment]
not the file is not in the classpath. :) ok :) so... I got a 🐛 :)

**Comment by SentryMan** [comment]
you can also try loading with the `-P` command line argument

**Comment by rob-bygrave** [comment]
Also note that the default loading (of application.properties|yaml) etc can be bypassed by using `Configuration.builder()` like:

https://github.com/avaje/avaje-config/blob/master/avaje-config/src/test/java/io/avaje/config/CoreConfigurationTest.java#L150-L154

BUT wait ...

> My use case is I have a single properties file that is application configuration but also ebean configuration

So what is the problem exactly? Ebean uses avaje-config by default so it isn't clear to me what the problem is?  You could have application and ebean configuration in the same place like application.properties - it isn't clear to me what the issue is here?

**Comment by rbygrave** [comment]
>  I'd like to avoid loading application.yaml file at all (and all other places that are possible) and load a specific file that is somewhere on the filesystem.

Right. So `System.setProperty("load.properties", fileName);` would **_almost_** work, in that currently (with that PR/fix) it would load that file **_but_** it currently would also by default load `application.yaml` etc and to get full control we don't want that.


> but does not allow me to load a different yaml file.

We can add that capability, so that we can do

```java
    var fileSource = new File("./src/test/resources/yaml/minimal.yaml");

    var configuration = Configuration.builder()
      .load(fileSource) // load from a file
      .load("hi.properties") // load from a resource
      .build();
```
... and this is the way to get full control over it. This is closer to what you were looking for?

**Comment by vnnv** [comment]
Yes, thanks :) it seems quite good to me :)

**Comment by rbygrave** [comment]
> Yes, thanks :) it seems quite good to me :)

Ok cool. I'll release that then as version 3.13 and it will also have the fix from @SentryMan, I'll do that now.

**Comment by vnnv** [comment]
:) Thank you Rob :) will update my deps :)

**Comment by rbygrave** [comment]
Ok, I'll close this as fixed in 3.13. Comment back here if it doesn't work as you expect etc. Cheers !!
